### ISSUE TITLE
Create a HBN datalad dataset

### ISSUE THREAD

**Issue by adswa** [datalad/datalad#6906]
@richford and I met at Neurohackademy, and he gave a very nice overview / data showcase on the HBN dataset. I learned that most of that dataset is openly available on S3 at ``s3://fcp-indi/data/Projects/HBN/BIDS_curated``. It should be easily possible to DataLaddify this similar to how we did it with the HCP dataset. Afterwards, on could use ``copy-file`` to create smaller dataset views, for example of each site, similar to how we subsampled e.g., the HCP movie data. Tagging @mattcieslak as well as an FYI.

**Comment by yarikoptic** [comment]
https://github.com/PennLINC/HBN_BIDS ?

**Comment by yarikoptic** [comment]
note: we should probably create some "README" or whatever repository within https://github.com/orgs/datalad-datasets/repositories to collect such issues which are not directly related to "DataLad core"

**Comment by adswa** [comment]
Thanks for the pointer! You're completely right, the work is already done. We'll look into doing this for the derivatives data then.

**Comment by yarikoptic** [comment]
FWIW I found some attempt (I think may be was done while talking to @satra and @djarecka on the topic) where I have used keen to them configuration for fmriprep, and Repronim/containers, and datalad-container.  FWIW here it is (without data obviously): https://github.com/yarikoptic/HBN_BIDS-fmriprep_try . But may be there is already precomputed derivatives based off this BIDS dataset already somewhere?

**Comment by richford** [comment]
For preprocessed fMRI, I'm not aware of any released work on this. But for preprocessed DWI and tractometry outputs, @adswa and I are working on DataLaddifying https://www.biorxiv.org/content/10.1101/2022.02.24.481303v1

**Comment by satra** [comment]
@smeisler and @jsmentch have a datalad dataset for hbn. we need to do a bit more curation and then i will post the dataset on some repo.

**Comment by yarikoptic** [comment]
> For preprocessed fMRI, I'm not aware of any released work on this. But for preprocessed DWI and tractometry outputs, @adswa and I are working on DataLaddifying https://www.biorxiv.org/content/10.1101/2022.02.24.481303v1

ah, cool. Meanwhile just for fun running

```shell
(git)smaug:/mnt/datasets/datalad/crawl-misc/hbn-bids-derivatives/qsiprep[master]
$> datalad crawl-init --save --template=simple_s3 bucket=fcp-indi prefix=data/Projects/HBN/BIDS_curated/derivatives/qsiprep/ to_http=1 
[INFO   ] Creating a pipeline for the fcp-indi bucket 

$> datalad crawl
[INFO   ] Loading pipeline specification from ./.datalad/crawl/crawl.cfg 
[INFO   ] Creating a pipeline for the fcp-indi bucket 
[INFO   ] Running pipeline [<datalad_crawler.nodes.s3.crawl_s3 object at 0x7f9f3a5add90>, sub(ok_missing=True, subs=<<{'url': {'^s3:++41 chars++/'}}>>), switch(default=None, key='datalad_action', mapping=<<{'commit': <fu++264 chars++0>>}>>, re=False)] 
[INFO   ] S3 session: Connecting to the bucket fcp-indi anonymously 
...
```
to see if that machinery still works . I hope that that dataset scale is ok for it to come out as a single dataset.

**Comment by yarikoptic** [comment]
unfortunately immediately ran into a problem of some files/versions being not allowed to be accessed:

```
$> datalad ls -aL s3://fcp-indi/data/Projects/HBN/BIDS_curated/derivatives/qsiprep/sub-NDARBD992CH7/anat/sub-NDARBD992CH7_desc-brain_mask.nii.gz
Connecting to bucket: fcp-indi
[INFO   ] S3 session: Connecting to the bucket fcp-indi anonymously 
Bucket info:
  Versioning: S3ResponseError: 403 Forbidden
     Website: S3ResponseError: 403 Forbidden
         ACL: S3ResponseError: 403 Forbidden
data/Projects/HBN/BIDS_curated/derivatives/qsiprep/sub-NDARBD992CH7/anat/sub-NDARBD992CH7_desc-brain_mask.nii.gz 2020-10-30T03:34:52.000Z 173346 ver:WaMdtau.Y5z1xZgB4.6fJcNc_IfCLFTj  acl:AccessDenied  http://fcp-indi.s3.amazonaws.com/data/Projects/HBN/BIDS_curated/derivatives/qsiprep/sub-NDARBD992CH7/anat/sub-NDARBD992CH7_desc-brain_mask.nii.gz?versionId=WaMdtau.Y5z1xZgB4.6fJcNc_IfCLFTj [E: 403]
```

<details>
<summary>that file is accessible without  versionId but that would make it "of not guaranteed version"</summary> 

```shell
smaug:/tmp
$> wget 'http://fcp-indi.s3.amazonaws.com/data/Projects/HBN/BIDS_curated/derivatives/qsiprep/sub-NDARBD992CH7/anat/sub-NDARBD992CH7_desc-brain_mask.nii.gz?versionId=WaMdtau.Y5z1xZgB4.6fJcNc_IfCLFTj'
--2022-08-01 15:32:39--  http://fcp-indi.s3.amazonaws.com/data/Projects/HBN/BIDS_curated/derivatives/qsiprep/sub-NDARBD992CH7/anat/sub-NDARBD992CH7_desc-brain_mask.nii.gz?versionId=WaMdtau.Y5z1xZgB4.6fJcNc_IfCLFTj
Resolving fcp-indi.s3.amazonaws.com (fcp-indi.s3.amazonaws.com)... 52.217.71.164
Connecting to fcp-indi.s3.amazonaws.com (fcp-indi.s3.amazonaws.com)|52.217.71.164|:80... connected.
HTTP request sent, awaiting response... 403 Forbidden
2022-08-01 15:32:39 ERROR 403: Forbidden.

(dev3) 1 6836 ->8.....................................:Mon 01 Aug 2022 03:32:39 PM EDT:.
smaug:/tmp
$> wget 'http://fcp-indi.s3.amazonaws.com/data/Projects/HBN/BIDS_curated/derivatives/qsiprep/sub-NDARBD992CH7/anat/sub-NDARBD992CH7_desc-brain_mask.nii.gz                                            
quote> 
(dev3) 1 6837 ->130.....................................:Mon 01 Aug 2022 03:32:43 PM EDT:.
smaug:/tmp
$> wget 'http://fcp-indi.s3.amazonaws.com/data/Projects/HBN/BIDS_curated/derivatives/qsiprep/sub-NDARBD992CH7/anat/sub-NDARBD992CH7_desc-brain_mask.nii.gz'
--2022-08-01 15:32:45--  http://fcp-indi.s3.amazonaws.com/data/Projects/HBN/BIDS_curated/derivatives/qsiprep/sub-NDARBD992CH7/anat/sub-NDARBD992CH7_desc-brain_mask.nii.gz
Resolving fcp-indi.s3.amazonaws.com (fcp-indi.s3.amazonaws.com)... 52.216.76.148
Connecting to fcp-indi.s3.amazonaws.com (fcp-indi.s3.amazonaws.com)|52.216.76.148|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 173346 (169K) [binary/octet-stream]
Saving to: ‘sub-NDARBD992CH7_desc-brain_mask.nii.gz’

sub-NDARBD992CH7_desc-brain_mask.nii.gz                          100%[=========================================================================================================================================================>] 169.28K  --.-KB/s    in 0.06s   

2022-08-01 15:32:46 (2.54 MB/s) - ‘sub-NDARBD992CH7_desc-brain_mask.nii.gz’ saved [173346/173346]

```
</details>

ping @engfranco (If I determined github handle correctly) ?! 

Meanwhile I will run `datalad ls` to see if there may be more of such 403s

**Comment by adswa** [comment]
Our approach right now is using the aws s3 client to list files and then hand them to ``addurl``


>  I hope that that dataset scale is ok for it to come out as a single dataset.

I think its pretty borderline large. @richford can correct me if I'm wrong but its around 700k files if I have seen it correctly.

**Comment by richford** [comment]
> @smeisler and @jsmentch have a datalad dataset for hbn. we need to do a bit more curation and then i will post the dataset on some repo.

@satra , did you mean that they are working on a preprocessed fMRI datalad dataset for HBN. Or that they also have another datalad dataset for "raw" HBN data. Or for preprocessed DWI?

**Comment by satra** [comment]
this is the raw dataset. and we plan to run fmriprep 22.0.0 which will generate freesurfer 7.2 + fmriprep derivatives and the latest mriqc which includes a few additional things.

steven may have additional thoughts on qsiprep output, which will also be run if a newer version is required/necessary.

**Comment by richford** [comment]
> Our approach right now is using the aws s3 client to list files and then hand them to `addurl`
> 
> > I hope that that dataset scale is ok for it to come out as a single dataset.
> 
> I think its pretty borderline large. @richford can correct me if I'm wrong but its around 700k files if I have seen it correctly.

We actually have two derivatives datasets that we are attempting to dataladdify:
- The preprocessed DWI dataset (i.e. QSIPrep outputs) has 81K files.
- The AFQ outputs have 706K files.

I think in both cases, we will construct sub-datasets for each participant.

**Comment by smeisler** [comment]
In regards to QSIPrep, the preprocessing in HBN-POD2 (https://www.biorxiv.org/content/10.1101/2022.02.24.481303v1.full) is good, even though it ran on an older version of QSIPrep. Post-processing reconstruction (e.g. tractography) could use newer versions to benefit from using FreeSurfer outputs. It is not a problem, at least now, if pre/post-processing use different QSIPrep versions, as they are largely independent workflows.

Since QSIPrep can now use FS in tractography, it might be nice to run fMRIPrep 22.0.0+ first, and then use the new FS 7.2 outputs in the QSIRecon workflows. I don't know how big an impact the surface ACT tractography would have on tract segmentations, if rerunning pyAFQ is a consideration.

**Comment by engfranco** [comment]
> unfortunately immediately ran into a problem of some files/versions being not allowed to be accessed:
> 
> ```
> $> datalad ls -aL s3://fcp-indi/data/Projects/HBN/BIDS_curated/derivatives/qsiprep/sub-NDARBD992CH7/anat/sub-NDARBD992CH7_desc-brain_mask.nii.gz
> Connecting to bucket: fcp-indi
> [INFO   ] S3 session: Connecting to the bucket fcp-indi anonymously 
> Bucket info:
>   Versioning: S3ResponseError: 403 Forbidden
>      Website: S3ResponseError: 403 Forbidden
>          ACL: S3ResponseError: 403 Forbidden
> data/Projects/HBN/BIDS_curated/derivatives/qsiprep/sub-NDARBD992CH7/anat/sub-NDARBD992CH7_desc-brain_mask.nii.gz 2020-10-30T03:34:52.000Z 173346 ver:WaMdtau.Y5z1xZgB4.6fJcNc_IfCLFTj  acl:AccessDenied  http://fcp-indi.s3.amazonaws.com/data/Projects/HBN/BIDS_curated/derivatives/qsiprep/sub-NDARBD992CH7/anat/sub-NDARBD992CH7_desc-brain_mask.nii.gz?versionId=WaMdtau.Y5z1xZgB4.6fJcNc_IfCLFTj [E: 403]
> ```
> 
> that file is accessible without versionId but that would make it "of not guaranteed version"
> ping @engfranco (If I determined github handle correctly) ?!
> 
> Meanwhile I will run `datalad ls` to see if there may be more of such 403s

Looks like all of 'data/Projects/HBN/BIDS_curated/*' was not created with the "read" for everyone. Changing that now. Guessing right now that the change should be completed in an hour or so. Let me know if you run into any more read issues.

**Comment by yarikoptic** [comment]
it is still having some file/versions not accessible, e.g.

```
$> datalad ls -aLr s3://fcp-indi/data/Projects/HBN/BIDS_curated/derivatives/qsiprep/sub-NDARBD992CH7/anat/sub-NDARBD992CH7_desc-brain_mask.nii.gz
Connecting to bucket: fcp-indi
[INFO   ] S3 session: Connecting to the bucket fcp-indi anonymously
Bucket info:
  Versioning: S3ResponseError: 403 Forbidden
     Website: S3ResponseError: 403 Forbidden
         ACL: S3ResponseError: 403 Forbidden
data/Projects/HBN/BIDS_curated/derivatives/qsiprep/sub-NDARBD992CH7/anat/sub-NDARBD992CH7_desc-brain_mask.nii.gz 2020-10-30T03:34:52.000Z 173346 ver:WaMdtau.Y5z1xZgB4.6fJcNc_IfCLFTj  acl:AccessDenied  http://fcp-indi.s3.amazonaws.com/data/Projects/HBN/BIDS_curated/derivatives/qsiprep/sub-NDARBD992CH7/anat/sub-NDARBD992CH7_desc-brain_mask.nii.gz?versionId=WaMdtau.Y5z1xZgB4.6fJcNc_IfCLFTj [E: 403]
```

I am running yet another greedy listing across the entire bucket to identify all of them (so far only on about 10% of that prefix) and will give you a full list later on if would be of help.

**Comment by engfranco** [comment]
Looks like the batch process crashed (way too many objects). I just restarted it now. If this doesn't work on each folder individually.

**Comment by yarikoptic** [comment]
@engfranco THANK YOU! I think we are almost there and I can detect only 1 prior version of a file to be not accessible:

```
smaug:/mnt/datasets/datalad/crawl-misc/hbn-bids-derivatives
$> datalad ls -aL s3://fcp-indi/data/Projects/HBN/BIDS_curated/derivatives/afq/sub-NDARAF440XWG/ses-HBNsiteRU/ROIs/sub-NDARAF440XWG_ses-HBNsiteRU_acq-64dir_space-T1w_desc-preproc_dwi_desc-ROI-PostParietal-2-includ
e.nii.gz
Connecting to bucket: fcp-indi
[INFO   ] S3 session: Connecting to the bucket fcp-indi anonymously 
Bucket info:
  Versioning: S3ResponseError: 403 Forbidden
     Website: S3ResponseError: 403 Forbidden
         ACL: S3ResponseError: 403 Forbidden
data/Projects/HBN/BIDS_curated/derivatives/afq/sub-NDARAF440XWG/ses-HBNsiteRU/ROIs/sub-NDARAF440XWG_ses-HBNsiteRU_acq-64dir_space-T1w_desc-preproc_dwi_desc-ROI-PostParietal-2-include.nii.gz 2022-04-09T03:11:48.000
Z 27201 ver:2om1_yevPkrJZGAQOeo23TsWJsSX75Fa  acl:AccessDenied  http://fcp-indi.s3.amazonaws.com/data/Projects/HBN/BIDS_curated/derivatives/afq/sub-NDARAF440XWG/ses-HBNsiteRU/ROIs/sub-NDARAF440XWG_ses-HBNsiteRU_ac
q-64dir_space-T1w_desc-preproc_dwi_desc-ROI-PostParietal-2-include.nii.gz?versionId=2om1_yevPkrJZGAQOeo23TsWJsSX75Fa [OK]
data/Projects/HBN/BIDS_curated/derivatives/afq/sub-NDARAF440XWG/ses-HBNsiteRU/ROIs/sub-NDARAF440XWG_ses-HBNsiteRU_acq-64dir_space-T1w_desc-preproc_dwi_desc-ROI-PostParietal-2-include.nii.gz 2022-04-09T03:11:42.000
Z 27201 ver:7E9qHFs_y2Eq9rPX1MrOMidCPIQNF62J  acl:AccessDenied  http://fcp-indi.s3.amazonaws.com/data/Projects/HBN/BIDS_curated/derivatives/afq/sub-NDARAF440XWG/ses-HBNsiteRU/ROIs/sub-NDARAF440XWG_ses-HBNsiteRU_ac
q-64dir_space-T1w_desc-preproc_dwi_desc-ROI-PostParietal-2-include.nii.gz?versionId=7E9qHFs_y2Eq9rPX1MrOMidCPIQNF62J [E: 403]
```

if you could make that 2nd older version accessible -- that would be great!

**Comment by satra** [comment]
@yarikoptic - are you creating a dataset or testing the penn linc one? the folder on fcp-indi may be missing certain data.

**Comment by yarikoptic** [comment]
> @yarikoptic - are you creating a dataset or testing the penn linc one? the folder on fcp-indi may be missing certain data.

If your are aware of missing data, could you please alert data submitters? I am just a brainless runner of datalad crawl ;-)

**Comment by satra** [comment]
> If your are aware of missing data, could you please alert data submitters? 

i don't know, but i saw some instances of anat missing in the bucket based on random spot checks, which i would not have expected.

@engfranco (fcp-indi) and @scovitz (pennlinc) i think would be the right people to know this. 

this is the bids datalad dataset we use: https://github.com/PennLINC/HBN_BIDS

**Comment by engfranco** [comment]
Hi Folks, just to be clear (and I apologize for the confusion). 
The dataset in s3://fcp-indi/data/Projects/HBN/BIDS_curated/  is mainly organized by @richford and is focused on preprocessing the DWI data. The raw data here is an early version of the curated data from pennlinc. 

The data in s3://fcp-indi/data/Projects/HBN/BIDS_datalad/ is being organized by PENNLINC ( @scovitz). This is more updated compared to 'BIDS_curated'. As far as I know, @satra is pulling data from here. 

In sum, if you just want HBN raw data, I would go with 'BIDS_datalad'. If you want DWI processed data, go with 'BIDS_curated"

**Comment by yarikoptic** [comment]
@engfranco could you please have a look at that [very last file version with 403ing permissions](https://github.com/datalad/datalad/issues/6906#issuecomment-1203057248)?

**Comment by engfranco** [comment]
> 

Done! 
Let me know if there are any more issues

**Comment by yarikoptic** [comment]
@engfranco my script failed on this one after fetching first ~200 files:

```shell
$> datalad ls -aL s3://fcp-indi/data/Projects/HBN/BIDS_curated/derivatives/qsiprep/sub-NDARYC594EHN/anat/sub-NDARYC594EHN_desc-brain_mask.nii.gz
Connecting to bucket: fcp-indi
[INFO   ] S3 session: Connecting to the bucket fcp-indi anonymously
Bucket info:
  Versioning: S3ResponseError: 403 Forbidden
     Website: S3ResponseError: 403 Forbidden
         ACL: S3ResponseError: 403 Forbidden
data/Projects/HBN/BIDS_curated/derivatives/qsiprep/sub-NDARYC594EHN/anat/sub-NDARYC594EHN_desc-brain_mask.nii.gz 2020-10-30T07:15:58.000Z 142420 ver:5Z7nbxYvqyH5vmG6w4yb3nOmlYxFSpS_  acl:AccessDenied  http://fcp-indi.s3.amazonaws.com/data/Projects/HBN/BIDS_curated/derivatives/qsiprep/sub-NDARYC594EHN/anat/sub-NDARYC594EHN_desc-brain_mask.nii.gz?versionId=5Z7nbxYvqyH5vmG6w4yb3nOmlYxFSpS_ [E: 403]
```
and I wondered how did I miss it... apparently `datalad ls` did crash with some connection error and we didn't retry :/  will follow up on that in https://github.com/datalad/datalad-deprecated/issues/61 and eventually hopefully would provide you a more complete list of problems.

**Comment by engfranco** [comment]
In theory, everything should be good to go now.

**Comment by richford** [comment]
Can I solicit some help here with my HBN datalad dataset. I'm making a datalad dataset for the HBN-POD2 project, encompassing the data at `s3://fcp-indi/data/Projects/HBN/BIDS_curated` with the derivative QSIPrep and pyAFQ output datasets as subdatasets. I think this would close this issue and is also needed to resolve an editorial request for publication of the HBN-POD2 manuscript.

After going through the instructions that @adswa gave me at Neurohackademy, I have the superdataset published at https://github.com/nrdg/HBN-POD2. However, when I clone that dataset with
```
datalad clone git@github.com:nrdg/HBN-POD2.git
```
First, I get the following INFO messages
```
[INFO   ] Unable to parse git config from origin
[INFO   ] X11 forwarding request failed
[INFO   ] [ERROR] [Errno 32] Broken pipe (BrokenPipeError)
[INFO   ] Remote origin does not have git-annex installed; setting annex-ignore
[INFO   ] This could be a problem with the git-annex installation on the remote. Please make sure that git-annex-shell is available in PATH when you ssh into the remote. Once you have fixed the git-annex installation, run: git annex enableremote origin
install(ok): /Users/richford/projects/neuro/hbn-pod2/datalad-dataset/HBN-POD2 (dataset)
```
But the dataset is cloned anyway so I carry on. However, when I try to get a subdataset, for example
```
cd HBN-POD2
datalad get sub-NDARAA075AMK
```
I get the following error
```
install(error): /Users/richford/projects/neuro/hbn-pod2/datalad-dataset/HBN-POD2/sub-NDARAA075AMK (dataset) [Failed to clone from any candidate source URL. Encountered errors per each url were:
- http://store.datalad.org/b46/e8587-dcdc-41ca-9cbd-882cef1fe9ab
  CommandError: 'git -c diff.ignoreSubmodules=none clone --progress http://store.datalad.org/b46/e8587-dcdc-41ca-9cbd-882cef1fe9ab /Users/richford/projects/neuro/hbn-pod2/datalad-dataset/HBN-POD2/sub-NDARAA075AMK' failed with exitcode 128 [err: 'Cloning into '/Users/richford/projects/neuro/hbn-pod2/datalad-dataset/HBN-POD2/sub-NDARAA075AMK'...
fatal: repository 'http://store.datalad.org/b46/e8587-dcdc-41ca-9cbd-882cef1fe9ab/' not found']
- http://store.datalad.org/b46/e8587-dcdc-41ca-9cbd-882cef1fe9ab/.git
  CommandError: 'git -c diff.ignoreSubmodules=none clone --progress http://store.datalad.org/b46/e8587-dcdc-41ca-9cbd-882cef1fe9ab/.git /Users/richford/projects/neuro/hbn-pod2/datalad-dataset/HBN-POD2/sub-NDARAA075AMK' failed with exitcode 128 [err: 'Cloning into '/Users/richford/projects/neuro/hbn-pod2/datalad-dataset/HBN-POD2/sub-NDARAA075AMK'...
fatal: repository 'http://store.datalad.org/b46/e8587-dcdc-41ca-9cbd-882cef1fe9ab/.git/' not found']
- git@github.com:nrdg/HBN-POD2.git/sub-NDARAA075AMK
  CommandError: 'git -c diff.ignoreSubmodules=none clone --progress git@github.com:nrdg/HBN-POD2.git/sub-NDARAA075AMK /Users/richford/projects/neuro/hbn-pod2/datalad-dataset/HBN-POD2/sub-NDARAA075AMK' failed with exitcode 128 [err: 'Cloning into '/Users/richford/projects/neuro/hbn-pod2/datalad-dataset/HBN-POD2/sub-NDARAA075AMK'...
X11 forwarding request failed
fatal: remote error:
  nrdg/HBN-POD2.git/sub-NDARAA075AMK is not a valid repository name
  Visit https://support.github.com/ for help
CommandError: 'ssh -o ControlPath=/Users/richford/Library/Caches/datalad/sockets/3c4a171b -o SendEnv=GIT_PROTOCOL [git@github.com](mailto:git@github.com) 'git-upload-pack '"'"'nrdg/HBN-POD2.git/sub-NDARAA075AMK'"'"''' failed with exitcode 1']]
```
I can verify that [store.datalad.org](http://store.datalad.org/) has data at http://store.datalad.org/b46/e8587-dcdc-41ca-9cbd-882cef1fe9ab/
But that doesn’t contain a .git folder so perhaps datalad does not see a repository there.

Can you help me understand what I've messed up here?

**Comment by mih** [comment]
Hey, sorry for the delay -- vacation, summer, beach, cocktails :upside_down_face: 

I looked at the example subdatasets that you have given, and I think the situation is that `create-sibling-ria` ran without the `--post-update-hook` flag. Its documentation is not very helpful:

```
--post-update-hook    Enable git's default post-update-hook for the created sibling.
```

but if given, it activates this hook in the repository on the server:

```sh
% cat hooks/post-update.sample
#!/bin/sh
#
# An example hook script to prepare a packed repository for use over
# dumb transports.
#
# To enable this hook, rename this file to "post-update".

exec git update-server-info
```

so basically, a `push` to this repo would trigger a `git update-server-info` run. And that not having happened is causing the exact issue you are seeing. Explanation from the manpage:

```
A dumb server that does not do on-the-fly pack generations must
have some auxiliary information files in $GIT_DIR/info and
$GIT_OBJECT_DIRECTORY/info directories to help clients discover
what references and packs the server has. This command generates
such auxiliary files.
```

I confirmed that these files are absent, hence I suspect this is what happened (not).

Please try rerunning the `create-sibling-ria` call you ran before (could not find it posted somewhere), and add the parameters
```
--existing reconfigure --post-update-hook
```
This should make the necessary modification and put you back in business.

Sorry that this was not more obvious -- neither the level of dumbness of the server, nor the semantics of `--post-update-hook` are adequately communicated. I will file an issue for the latter.

**Comment by yarikoptic** [comment]
FWIW -- how large is the qsiprep -- so far I see over 1TB fetched and it is still going ;)

**Comment by richford** [comment]
The curated BIDS, QSIPrep outputs, and pyAFQ output combined come out to 3.3 TiB as verified by
```
aws s3 ls s3://fcp-indi/data/Projects/HBN/BIDS_curated --recursive --human-readable --summarize
```

The QSIPrep outputs alone are 1.4 TiB and the AFQ outputs alone are 1.6 TiB.

**Comment by richford** [comment]
@mih Thanks for your quick reply! I tried your suggestion and I think we have progress but not success. I can now `datalad get` various participants, e.g.,
```
> datalad clone git@github.com:nrdg/HBN-POD2.git
[INFO   ] Unable to parse git config from origin
[INFO   ] X11 forwarding request failed
[INFO   ] [ERROR] [Errno 32] Broken pipe (BrokenPipeError)
[INFO   ] Remote origin does not have git-annex installed; setting annex-ignore
[INFO   ] This could be a problem with the git-annex installation on the remote. Please make sure that git-annex-shell is available in PATH when you ssh into the remote. Once you have fixed the git-annex installation, run: git annex enableremote origin
install(ok): /Users/richford/projects/neuro/hbn-pod2/datalad-dataset/HBN-POD2 (dataset)
```
Followed by a `datalad get` for one of the participants.
```
> cd HBN-POD2
> datalad get -n sub-NDARHM611GDR
install(ok): /Users/richford/projects/neuro/hbn-pod2/datalad-dataset/HBN-POD2/sub-NDARHM611GDR (dataset) [Installed subdataset in order to get /Users/richford/projects/neuro/hbn-pod2/datalad-dataset/HBN-POD2/sub-NDARHM611GDR]
```
However, when I navigate to the subdataset
```
cd sub-NDARHM611GDR
ls -al
```
I get only
```
total 4
drwxr-xr-x    5 richford staff   160 Aug 18 15:07 ./
drwxr-xr-x 2622 richford staff 83904 Aug 18 15:05 ../
drwxr-xr-x    4 richford staff   128 Aug 18 15:07 .datalad/
drwxr-xr-x   14 richford staff   448 Aug 18 15:07 .git/
-rw-r--r--    1 richford staff    55 Aug 18 15:07 .gitattributes
```

So it would appear I've still done something wrong.

**Comment by richford** [comment]
@mih, you mentioned that I never posted the datalad commands that I used. Here's my best recollection of the commands (using bash history for reference).

First, I created a file at `.datalad/procedures/cfg_hbn.sh`
```
set -e -u

cd "$1"
echo "Enable special remote at $1"
git annex initremote datalad type=external externaltype=datalad encryption=none
autoenable=true || git annex enableremote datalad
git annex enableremote datalad autoenable=true
```
Followed by these commands in the superdataset root
```
datalad save -m "Add remote config file to .datalad/procedures" --to-git
datalad run-procedure --discover

# Use datalad addurls to index the dataset (this one took quite a while)
cat ~/all_subs.json | jq --slurp . | datalad addurls -c hbn --exclude-autometa '*' - 'http://fcp-indi.s3.amazonaws.com/{url}' '{target}' --drop-after --nosave
datalad save

# Add configuration to superdataset
git config -f .datalad/config "datalad.get.subdataset-source-candidate-300origin" "ria+http://store.datalad.org#{id}"
datalad save

# Publish to store.datalad.org
datalad create-sibling-ria -s ria-store --storage-sibling off --recursive ria+ssh://store.datalad.org/srv/store.datalad.org/www
datalad push --to ria-store --recursive``

# Publish superdataset to GitHub
datalad create-sibling-github --access-protocol ssh --credential github-pat -s github nrdg/HBN-POD2
datalad push --to github
```
where the file `~/all_subs.json` contains the S3 information for each file. Here are the last ten lines for example
```
{"url":"data/Projects/HBN/BIDS_curated/sub-ndarvn280jtn/ses-HBNsiteCBIC/dwi/sub-ndarvn280jtn_ses-HBNsiteCBIC_acq-64dir_dwi.json","target":"sub-ndarvn280jtn//ses-HBNsiteCBIC/dwi/sub-ndarvn280jtn_ses-HBNsiteCBIC_acq-64dir_dwi.json","etag":"\"f517e5e5f204ee04839de8b2e37cc8fd\""}
{"url":"data/Projects/HBN/BIDS_curated/sub-ndarvn280jtn/ses-HBNsiteCBIC/dwi/sub-ndarvn280jtn_ses-HBNsiteCBIC_acq-64dir_dwi.nii.gz","target":"sub-ndarvn280jtn//ses-HBNsiteCBIC/dwi/sub-ndarvn280jtn_ses-HBNsiteCBIC_acq-64dir_dwi.nii.gz","etag":"\"15cf2b41ac9d8930571a91949b03e430-16\""}
{"url":"data/Projects/HBN/BIDS_curated/sub-ndarvn280jtn/ses-HBNsiteCBIC/fmap/sub-ndarvn280jtn_ses-HBNsiteCBIC_acq-dwi_dir-AP_epi.json","target":"sub-ndarvn280jtn//ses-HBNsiteCBIC/fmap/sub-ndarvn280jtn_ses-HBNsiteCBIC_acq-dwi_dir-AP_epi.json","etag":"\"f83a82c1fcf9b4d50aeb2fa826c0465f\""}
{"url":"data/Projects/HBN/BIDS_curated/sub-ndarvn280jtn/ses-HBNsiteCBIC/fmap/sub-ndarvn280jtn_ses-HBNsiteCBIC_acq-dwi_dir-AP_epi.nii.gz","target":"sub-ndarvn280jtn//ses-HBNsiteCBIC/fmap/sub-ndarvn280jtn_ses-HBNsiteCBIC_acq-dwi_dir-AP_epi.nii.gz","etag":"\"6d66c63e4f5d010bddebf96c1a118bec\""}
{"url":"data/Projects/HBN/BIDS_curated/sub-ndarvn280jtn/ses-HBNsiteCBIC/fmap/sub-ndarvn280jtn_ses-HBNsiteCBIC_acq-dwi_dir-PA_epi.json","target":"sub-ndarvn280jtn//ses-HBNsiteCBIC/fmap/sub-ndarvn280jtn_ses-HBNsiteCBIC_acq-dwi_dir-PA_epi.json","etag":"\"dd7444b8525e1973a1b47c652b1cb843\""}
{"url":"data/Projects/HBN/BIDS_curated/sub-ndarvn280jtn/ses-HBNsiteCBIC/fmap/sub-ndarvn280jtn_ses-HBNsiteCBIC_acq-dwi_dir-PA_epi.nii.gz","target":"sub-ndarvn280jtn//ses-HBNsiteCBIC/fmap/sub-ndarvn280jtn_ses-HBNsiteCBIC_acq-dwi_dir-PA_epi.nii.gz","etag":"\"fc6db01cf9eaf2c0aa764625ff264faf\""}
{"url":"data/Projects/HBN/BIDS_curated/sub-ndarvn280jtn/ses-HBNsiteCBIC/fmap/sub-ndarvn280jtn_ses-HBNsiteCBIC_acq-fMRI_dir-AP_epi.json","target":"sub-ndarvn280jtn//ses-HBNsiteCBIC/fmap/sub-ndarvn280jtn_ses-HBNsiteCBIC_acq-fMRI_dir-AP_epi.json","etag":"\"492caf222db12eac7b5a6628a3fd6601\""}
{"url":"data/Projects/HBN/BIDS_curated/sub-ndarvn280jtn/ses-HBNsiteCBIC/fmap/sub-ndarvn280jtn_ses-HBNsiteCBIC_acq-fMRI_dir-AP_epi.nii.gz","target":"sub-ndarvn280jtn//ses-HBNsiteCBIC/fmap/sub-ndarvn280jtn_ses-HBNsiteCBIC_acq-fMRI_dir-AP_epi.nii.gz","etag":"\"c8a972b642bc0df7af13629d0f880529\""}
{"url":"data/Projects/HBN/BIDS_curated/sub-ndarvn280jtn/ses-HBNsiteCBIC/fmap/sub-ndarvn280jtn_ses-HBNsiteCBIC_acq-fMRI_dir-PA_epi.json","target":"sub-ndarvn280jtn//ses-HBNsiteCBIC/fmap/sub-ndarvn280jtn_ses-HBNsiteCBIC_acq-fMRI_dir-PA_epi.json","etag":"\"548939b8734449883154a10834a0024b\""}
{"url":"data/Projects/HBN/BIDS_curated/sub-ndarvn280jtn/ses-HBNsiteCBIC/fmap/sub-ndarvn280jtn_ses-HBNsiteCBIC_acq-fMRI_dir-PA_epi.nii.gz","target":"sub-ndarvn280jtn//ses-HBNsiteCBIC/fmap/sub-ndarvn280jtn_ses-HBNsiteCBIC_acq-fMRI_dir-PA_epi.nii.gz","etag":"\"4e67ee8755cc410715e7e03b94562feb\""}
```

**Comment by mih** [comment]
Hey, great that there was progress!

Can you compare the state of the clone (`git show`) with what you'd expect it to be?

I suspect the residual issue is a mere artifact of us having enabled the hook only after the `push`. I would hope that a `datalad push -f gitpush` would be able to rectify that.

**Comment by richford** [comment]
My apologies for the delayed update. I tried your suggestion @mih to no avail. But then I thought more about what @yarikoptic said to me over email and he convinced me that having a subdataset for each participant was probably overkill. So I now have just the main superdataset and one subdataset per derivative (i.e., one for QSIPrep and one for AFQ). With only three datasets, I was happy to do `datalad push -r --to github` and have three GitHub repositories for this project:

- https://github.com/nrdg/HBN-POD2
- https://github.com/nrdg/HBN-POD2-derivatives-qsiprep
- https://github.com/nrdg/HBN-POD2-derivatives-afq

where the latter two are also subdatasets in the derivatives folder of the first one.

I think this will satisfy the SciData editors. So I think we can close this issue.

There is, however, some clean up that I should do. As a result of my failed attempts above, there are a lot of files on store.datalad.org that we no longer need. @yarikoptic or @mih, are there any helpful datalad commands that I should use to clean up my mess? Or would it suffice to delete all of the files in /srv/www/store.datalad.org for which I am the owner?
